tests/cross-compile/cpp-resource.slang

```diff
@@ -1,4 +1,4 @@
-//TEST:CPP_COMPILER_COMPILE: -profile cs_5_0 -entry computeMain -target cpp
+//TEST(shared-library):CPP_COMPILER_COMPILE: -profile cs_5_0 -entry computeMain -target callable
```

Comment:
I get that we need to test this path, but I honestly can't tell why we'd support `-target callable` as a thing. It seems 100% redundant with `-target dll`.

Reply:
One unrelated thing that we'll need to deal with sooner or later: we really want to support compiling a file with N entry points down to a *single* DLL. That is, the following should probably be legal:

```
slangc mystuff.slang -o mystuff.dll
```

As a user I'd expect that to find *all* the entry points specified in `mystuff.slang` (using `[shader("...")]`) and then compile a DLL with each of those as an entry point. We don't currently have a lot of support in Slang for specifying an output file to be used for an entire target, as opposed to per-entry-point within a target, so that would need to be addressed first.

Realistically we can/should support specifying a single output file for a translation unit with multiple entry points for DXIL and SPIR-V too, so if we can flush the pipe for that support one one target it might be able to benefit multiple of them.

(All of this is leaving aside the looming issues around what the most desirable "do what I mean" semantics for the CPU target should be)

Reply:
`-target callable` whilst not super useful is handy in so far as it means the product produced is loadable and thus more likely a valid thing. There is a test that is written just that way for this reason. 
This would be less of an issue if we didn't have the slangc work as if it is a separate process. But you know just doing it from the command line to test - is useful imho. 

On compiling a file with multiple entry points into a single DLL - yep absolutely.